homeassistant/components/spider/config_flow.py

```diff
@@ -0,0 +1,67 @@
+"""Config flow for Spider."""
+import logging
+
+from spiderpy.spiderapi import SpiderApi, SpiderApiException, UnauthorizedException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SpiderConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Spider config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the Spider flow."""
+        self.data = {
+            CONF_USERNAME: "",
+            CONF_PASSWORD: "",
+            CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL,
+            "login_response": None,
+        }
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        errors = {}
+        if user_input is not None:
+            self.data[CONF_USERNAME] = user_input["username"]
+            self.data[CONF_PASSWORD] = user_input["password"]
+
+            if CONF_SCAN_INTERVAL in user_input:
+                self.data[CONF_SCAN_INTERVAL] = user_input["scan_interval"]
```

Comment:
Ideally as developers we choose an a good scan interval and we don't expose an option to configure it as most users won't know what a good value is.  If we allow them to choose one in the UI, they may choose one that overloads cloud servers which can result in service provider rate-limiting or taking adverse action against which affects all users.

If we really need to adjust the scan interval, we put scan intervals in an options flow.

Reply:
Thank you and you are totally right. Should I also make it static then within the code rather then configurable like it is here? https://github.com/home-assistant/core/blob/dev/homeassistant/components/spider/__init__.py#L26

Reply:
I'd remove it from the config schema so it doesn't show up for new users, but keep it for the import and use the default when its not defined during the entry setup.

Reply:
Sorry for the delay. I believe I changed it in the way you requested it. Let me know if this is up to standards. 

Next steps after this MR is to make sure it's async.